app/services/sarses_generator.py

```diff
@@ -0,0 +1,135 @@
+# pylint: disable=no-member
+
+"""This module contains logic for composing marketplace DB dump,
+    recommendations and user actions into SARSes
+    needed for Recommender system's replay buffer.
+
+    marketplace DB dump ---\
+                            \
+    recommendations ---------+----> SARSes
+                            /
+    user actions ----------/
+
+    It assumes that above data are stored in database before call.
+    """
+
+import itertools
+from app.db.mongo_models import UserAction, Recommendation, State, Sars
+from app.reward_mapping import ua_to_reward_id
+
+
+RECOMMENDATION_PAGES_IDS = ["catalogue_services_list"]
```

Comment:
Constant lists (especially if they're package / globally scoped) should be tuples:

 RECOMMENDATION_PAGES_IDS = ("catalogue_services_list", )

This way they could not be *accidentally* modified somewhere in the code